src/main/kotlin/no/nav/amt/deltaker/bff/deltaker/DeltakerService.kt

```diff
@@ -134,28 +135,46 @@ class DeltakerService(
             status = status,
         )
 
-        upsert(deltaker)
+        upsertKladd(deltaker)
 
         return deltakerRepository.get(deltaker.id).getOrThrow()
     }
 
-    fun oppdaterDeltaker(deltakeroppdatering: Deltakeroppdatering) {
+    fun oppdaterDeltaker(
+        deltakeroppdatering: Deltakeroppdatering,
+        personIdent: String,
+        deltakerlisteId: UUID,
+    ) {
+        if (deltakeroppdatering.status.type in AKTIVE_STATUSER && harEndretStatus(deltakeroppdatering)) {
+            val tidligereDeltakelser = getDeltakelser(personIdent, deltakerlisteId).filter {
+                it.id != deltakeroppdatering.id && it.harSluttet()
+            }
+            tidligereDeltakelser.forEach {
+                deltakerRepository.settKanIkkeEndres(it.id)
```

Comment:
Dette kan gjøres med en update i steden for mange ved en `update where id in ()`